index.js

```diff
@@ -46,7 +48,12 @@ class PCancelable {
 					throw new Error('The `onCancel` handler was attached after the promise settled.');
 				}
 
-				this._cancelHandlers.push(handler);
+				const fn = (...args) => {
```

Comment:
Can you undo this change? A much better solution is to move the `this._isCanceled = true;` before the if here:

https://github.com/jonschlinkert/p-cancelable/blob/deaeee112adc3d105aaec2074de971af5dd48c3a/index.js#L90-L100

Reply:
https://github.com/sindresorhus/p-cancelable/commit/4b7d014c5be44889e40014dea098ae44819b95d6